java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -81,9 +83,9 @@ class EntityHandleManager(
         storageKey: StorageKey,
         ttl: Ttl = Ttl.Infinite,
         idGenerator: Id.Generator = Id.Generator.newSession()
-    ): Handle =
+    ): SingletonHandle<T> =
```

Comment:
I think the problem you're seeing with this is that `SingletonHandle` is not a supertype to any of the types of objects returned in the `let`.    It might be worth switching back to `Handle`, but also: parametrizing `Handle` on `T`.

Reply:
Hmm, then we have two problems. `SingletonHandle` needs to be a super type of all of the objects returned in the let (if I understand the 'by' correctly).

Aren't we providing the implementation of the interfaces `ReadSingletonHandle`, `WriteSingletonHandle` and `ReadWriteSingletonHandle` using singletonHandle as the source?

Haven't been able to find out what the `object : `...`{}` parts do though.

Reply:
Back when HandleMode was actually passed as a param directly, you may have been able to use type capture to have the return type be inferred, e.g. 

```
sealed class HandleMode<E : Entity, S : Handle<E>, C : Handle<E> {
   class Read : HandleMode<E, ReadSingletonHandle<E>, ReadCollectionHandle<E>
}

fun <E : Entity, S : Handle<E>, C : Handle<E>> createCollectionHandle(..., mode: HandleMode): C { 

}
```

The idea here being that when you invoked createCollectionHandle(..., Read), you'd have S = ReadSingletonHandle<*>, and C = ReadCollectionHandle<*> captured from the type parameters of the mode. 

However, now that this has been hidden in HandleSpec, it's not feasible.